### DISCUSSION TITLE
Why can't I push Structural datas to Sap2000 using Push component?

### DISCUSSION THREAD

**Discussion by bearlove163** [BHoM/BHoM#1569] [General]
I have used SAP2000Adapter to communicate with Sap2000. I can use Execute components to operations Sap2000 such as New and RUN order, but I cannot enter data from GH into Sap2000using Push component.
Can you tell me how I can import data into Sap2000.
Thank you.

Error information:“1. This component failed to run properly.
- Error: 返回参数具有无效的类型。
- Occured in BH.UI.Base.Caller.Run(List`1 inputs)
     called fromBH.UI.Base.Caller.Run()
- Are you sure you have the correct type of inputs? Check their description for more details.”

**Comment by FraserGreenroyd** [comment]
Hi @bearlove163 , welcome to BHoM!

Would you be able to translate the error message for us? I've run it through Google Translate and it says `Return parameter has invalid type` which doesn't match any errors in BHoM that I can find, so I'm wondering if the translation is a little off.

Would you be able to provide a sample script as well? @peterjamesnugent probably god to get you involved on this as you're more likely to know about SAP2000 than I am?

**Reply by bearlove163** [reply]
I use SapAdapte component link with sap2000 successfully, and I can transfer bars that adopt default sectionProperty into Sap2000 using Push component.
![image](https://github.com/BHoM/BHoM/assets/87690426/f7d1a8ad-1fa5-47a8-87ba-69bb19a65391)
However I can't transfer bars into Sap2000 using Push component when parameter of sectionProperty in bar component is used custom sectionProperty.
![image](https://github.com/BHoM/BHoM/assets/87690426/9fe497a3-f6ec-4e80-90c4-69e6bf2807d1)

Could you tell me how I can deal with above error?

**Reply by peterjamesnugent** [reply]
Hi @bearlove163, SAP2000 does not appear to implement FreeformProfiles:
https://github.com/BHoM/SAP2000_Toolkit/blob/develop/SAP2000_Adapter/CRUD/Create/Section.cs

If you're familiar with SAP2000 API, would you be interested in adding this functionality?

**Comment by bearlove163** [comment]
您的邮件已收到。

**Comment by peterjamesnugent** [comment]
Hi @bearlove163, it depends on what objects you are trying to push. As a first pass, it may be worth looking at the `Create` methods available within the SAP2000_Toolkit: https://github.com/BHoM/SAP2000_Toolkit/tree/develop/SAP2000_Adapter/CRUD/Create
